apheleia.el

```diff
@@ -389,18 +389,29 @@ as its sole argument."
                       ;; changes, and 2 if error.
                       (memq status '(0 1)))))))
 
-(defun apheleia--run-formatter (command callback)
-  "Run a code formatter on the current buffer.
-The formatter is specified by COMMAND, a list of strings or
-symbols (see `apheleia-format-buffer'). Invoke CALLBACK with one
-argument, a buffer containing the output of the formatter.
-
-If COMMAND uses the symbol `file' and the current buffer is
-modified from what is written to disk, then don't do anything."
+(defun apheleia--format-command (command &optional stdin)
+  "Format COMMAND into a shell-ccmd and list of file paths.
+Should return a list with the car being the optional input file-name
+the cadr being the optional output file-name and the cddr being the
+cmd to run.
+
+STDIN is the optional buffer to use when creating a temporary file for
+the formatters standard input.
+
+If COMMAND uses the symbol `file' and the current buffer is modified
+from what is written to disk, then return nil meaning meaning no
+cmd is to be run."
+  ;; WARN: The condition with regards to `file' only checks the original
+  ;; file so it doesn't apply the formats from multiple formatters being
+  ;; run. For example if formatter 1 produces some output and formatter
+  ;; 2 modifies file in place then the changes from formatter 1 may not
+  ;; apply at all.
```

Comment:
Ideally, we can handle all supported features for formatter chaining. But failing that, I think it would be good enough to report an error or warning when we hit an unsupported case.